File: ABC/178/e.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
  int N;
  cin >> N;
  vector<int> x(N);
  vector<int> y(N);
  int i;
  for(i = 0; i < N; i++){
    cin >> x.at(i) >> y.at(i);
  }

  vector<vector<int>> d(N, vector<int>(N));
  int j;
  for(i = 0; i < N; i++){
    for(j = 0; j < N; j++){
      if(i < j){
        d.at(i).at(j) = abs(x.at(i)-x.at(j)) + abs(y.at(i)-y.at(j))
      }
    }
  }

  max(max(d)) // Is it correct??
}

File: ABC/178/e.py
N = int(input())
x = [0 for i in range(N)]
y = [0 for i in range(N)]
for i in range(N):
  x[i], y[i] = map(int, input().split())

d = [[0 for i in range(N)] for j in range(N)]
for i in range(N):
  for j in range(N):
    if i < j:
      d[i][j] = abs(x[i]-x[j]) + abs(y[i]-y[j])
print(max(max(d)))

# As you know, returned "TLE"...